### PR TITLE
Docker: use more OS packages for dependencies

### PR DESCRIPTION
**Pull request by hainest**
This should have been part of #1211

### CODE REVIEW

**Comment by hainest** [comment]
> Does slack external find it work to find those?

No. It won't find libraries- just executables in $PATH. :(

**Comment by vsoch** [comment]
Any reason to printf into the file vs. storing a packages.yaml alongside the container?

**Comment by hainest** [comment]
None other than I didn't know you could do that. If we have both a hand-written packages.yaml and are using `spack external find`, where would the best place be to keep the packages.yaml?

**Comment by vsoch** [comment]
If you are just building the container, you'd keep it in the docker/ folder with COPY to where it needs to be and add it before you setup the environment.

**Comment by hainest** [comment]
@vsoch I was able to get two of the three added with `spack find external`. The last one I'll leave for now. You, @kupsch, @bigtrak, and I should have a chat (preferably in the soon-ish timeframe) to expand what we have now to support multiple OSes and what we want to do about version evolution for dependencies. I'd also like to think about abstracting the base containers into their own layers, if that makes sense for whatever we decide on an overarching architecture.

**Comment by vsoch** [comment]
@hainest could you temporarily add the PR trigger to the workflow to build the new container (and double check it won't deploy) so we can be 100% sure it builds ok? once that is done ping me to take a look, then we can undo that change and merge.

**Comment by hainest** [comment]
Did I do that correctly?

File: .github/workflows/base-containers.yaml [hunk 1]
```diff
@@ -14,6 +14,10 @@ on:
   push: 
     branches:
       - master
+  
+  pull_request:
```
Comment (vsoch) [inline .github/workflows/base-containers.yaml:18]: Yep! If you are lazy in the future you can just do:
```yaml
pull_request: []
```

**Comment by vsoch** [comment]
@hainest looks good! Just remove the PR trigger and +1 to merge.

52 minutes - Dyninst you CHONK!

**Comment by hainest** [comment]
52 minutes?! That's a paddlin'.

**Pull request merged**